fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/RankedSetIndexHelper.java

```diff
@@ -50,6 +54,54 @@
 public class RankedSetIndexHelper {
     public static final Tuple COMPARISON_SKIPPED_SCORE = Tuple.from(Comparisons.COMPARISON_SKIPPED_BINDING);
 
+    public static final RankedSet.HashFunction MURMUR3_HASH_FUNCTION = new GuavaHashFunction(Hashing.murmur3_32());
+
+    /**
+     * Use {@code com.google.common.hash.HashFunction}s as {@code RankedSet.HashFunction}s.
+     */
+    public static class GuavaHashFunction implements RankedSet.HashFunction {
+        @Nonnull
+        private final HashFunction guava;
+
+        public GuavaHashFunction(HashFunction guava) {
+            this.guava = guava;
+        }
+
+        @Override
+        public int hash(byte[] key) {
+            return guava.hashBytes(key).asInt();
+        }
+    }
+
+    /**
+     * Known hash functions available as index options.
+     */
+    public enum HashFunctionNames {
```

Comment:
As this enum is in an `INTERNAL` class, it's technically "unadvised" for users to use this, but the alternative is they need their own constants in their code if they want to use the names here. I suppose we could leave it here if we're not "entirely happy" with this API (yet?), or move it out.

Reply:
I made this top level. Since that meant it should own the other statics and enum init order is fixed, I needed to also make it a regular class.